### ISSUE TITLE
asymmetric neck shape

### ISSUE THREAD

**Issue by zu4lu** [mnesarco/MarzWorkbench#23]
Hi Frank, last version of maz is very good, but i'm missing one feature badly, the asymmetric neck shapes(if you ever played one of these, you know what i mean). Is this possible?
Michael

**Comment by mnesarco** [comment]
Asymetric necks are supported in the c++ in-development version. But it is not public yet.

**Comment by mnesarco** [comment]
https://forum.freecad.org/viewtopic.php?t=58729&start=10#p505612

**Comment by mnesarco** [comment]
Thank you. 
The best help I need is in cross platform packaging of the c++ version for distribution.

- Linux: .AppImage  (I can do that)
- Mac: .app (I have no idea)
- Windows: .msi or portable .exe (I have not used Windows for decades... so I have no idea)

I have no timeline, I do this in spare time.

![image](https://user-images.githubusercontent.com/1087317/229644300-936d4048-7b13-4e23-9ba2-d76e81ee5ee5.png)

I have not published it yet because there are many things pending to package this as a product.

**Comment by mnesarco** [comment]
@zu4lu 
Yes in the current WB it is possible to define a custom neck profile with a single asymmetry but not compound asymmetry.

https://github.com/mnesarco/MarzWorkbench/wiki/Neck-Profile-Specs

**Comment by zu4lu** [comment]
Thank you, will you make it possible it in the workbench?

**Comment by mnesarco** [comment]
> Thank you, will you make it possible it in the workbench?

Maybe.

**Comment by mnesarco** [comment]
I am working on an editor for neck profiles. It will support double asymmetric necks and it will be available in the FreeCAD version.

**Comment by zu4lu** [comment]
Hi, great news, i'm looking forward to it

Am Montag, Mai 06, 2024 17:59 CEST, schrieb Frank David Martínez M ***@***.***>:
 
 
I am working on an editor for neck profiles. It will support double asymmetric necks and it will be available in the FreeCAD version.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mnesarco/MarzWorkbench/issues/23#issuecomment-2096390184", "url": "https://github.com/mnesarco/MarzWorkbench/issues/23#issuecomment-2096390184", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]***>
-- 
lg
Michael Kaufmann

**Comment by zu4lu** [comment]
Hi Frank, maybe this will help. I make my shapes based on this and its fantastic.
https://www.kappi.com/blog/2012/10/asymmetric-neck-shapes/

**Comment by mnesarco** [comment]
@zu4lu thank you for the link, but i already have developed the double asymmetric neck editor in my private c++ version.  I just have had not time to port it here. The idea is actually pretty similar:

![image](https://github.com/mnesarco/MarzWorkbench/assets/1087317/baee61d2-14c9-4240-8355-dfe96d79dbf6)

**Comment by zu4lu** [comment]
good news, thanks

**Comment by mnesarco** [comment]
> Oh wow, would it be possible to desig an own shape like an enduro neck?

Not currently. There are infinite variations for each aspect of a guitar, it is very hard to cover all possible cases. Said that, it is not that difficult to make the custom neck profile more flexible, but it also makes the UI more complex. The idea of this Workbench is to cover common cases for amateur guitar makers and for experimentation. Automation will never cover all the Artistic or advanced features required by professionals.

**Comment by Cysign** [comment]
That makes sense. I'd still like to build my first homemade guitar with an Enduro neck (since I can't afford a Boden Prog NX 7 - just to see if I like it :D ). So I have to get as close as possible to that design and to the rest by hand. Thanks a lot for all your great work in this project <3
